src/sparseml/optim/helpers.py

```diff
@@ -219,13 +219,98 @@ def update_recipe_variables(recipe_yaml_str: str, variables: Dict[str, Any]) ->
     return rewrite_recipe_yaml_string_with_classes(container)
 
 
+def print_staged_container(stage_container):
+    for k, v in stage_container.items():
+        print("===========")
+        print(f"KEY:\n{k}\n")
+        print("VALUES:\t")
+
+        if type(v) != dict:
+            print(v)
+        else:
+            for _k, _v in v.items():
+                if type(_v) == list:
+                    print(f"{_k}:")
+                    print("[")
+                    for i in _v:
+                        print(f"\t{i}")
+                    print("]")
+                else:
+                    raise ValueError("We should not get here")
+
+
+def check_if_staged_recipe(container: Dict) -> Tuple[Dict, bool]:
+    """
+    Check whether the container has been generated by the standard recipe
+    or staged recipe file. If the latter case, transforms the container
+    to following structure:
+
+    {
+    'global_variables: [{var1: 'foo'}, {var2: 'fizz'}, ...]
+    'stage alpha': {
+        'stage_variables': [{'num_epochs': 100}, {'init_lr': 0.1}]
+        'modifiers': ['OBJECT.EpochRangeModifier': {...},
+                     ['OBJECT.LearningRateFunctonModifier': {...},
+                      ...]
+    'stage beta': {
+        'stage_variables': [{'num_epochs': 50}, {'init_lr': 0.2}]
+        'modifiers': ['OBJECT.EpochRangeModifier': {...},
+                     ['OBJECT.LearningRateFunctonModifier': {...},
+                      ...]
+    ...
+    }
+
+    Returns a container dictionary and a boolean (staged recipe or not).
+    """
+    staged_recipe = True
+    stage_signature = "_stage"
+    modifier_signature = "_modifier"
+
+    if not any([stage_signature in key for key in container.keys()]):
+        # if there are no stages in the container, this must be standard recipe, pass
+        return container, not staged_recipe
+
+    else:
+        # parsing the container to create stage_container
+        stage_container = {"global_variables": []}
```

Comment:
what is the goal for structure and usage of `stage_container` here?

Reply:
Regarding the goal for structure, I am still not sure how do we treat `global_variables` and "`stage_variables`".

From our discussion yesterday I understand that we want to break down the staged recipe into a single list of modifiers. Those modifiers will get activated/deactivated according to e.g. their start/end epoch (their order in the compound sparsity training routine). 

This means that the YAML definitions of a modifier in the list should probably be in its' final stage: by the time it gets passed to the manager, it should have evaluated modifier arguments incl. appropriate treatment of `global_variables` and `stage_variables`. 

The pending question is: given, that our goal structure of `staged_container` is a dictionary {`stage_name`: `List[modifiers]`}, should we include the variables (both global and stage variables) into their respective modifiers? I'd love to sync with you on that one.

Reply:
Let's maybe discuss this more holistically today, so I have bird's-eye view over the mechanism ok?